src/components/Popup/Popup.tsx

```diff
@@ -2,73 +2,78 @@ import * as React from 'react'
 import * as PropTypes from 'prop-types'
 import _ from 'lodash'
 import { Popper, PopperChildrenProps } from 'react-popper'
-import rtlCSSJS from 'rtl-css-js'
 
-import { childrenExist, customPropTypes, UIComponent, IRenderResultConfig } from '../../lib'
+import { childrenExist, AutoControlledComponent, IRenderResultConfig } from '../../lib'
 import { ItemShorthand, Extendable, ReactChildren } from '../../../types/utils'
-import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
-import Portal, { TriggerAccessibility } from '../Portal'
+
+import Ref from '../Ref'
+import computePopupPlacement, { Alignment, Position } from './positioningHelper'
+
 import PopupContent from './PopupContent'
+
 import { PopupBehavior } from '../../lib/accessibility'
 import {
   Accessibility,
   AccessibilityActionHandlers,
   IAccessibilityBehavior,
 } from '../../lib/accessibility/interfaces'
-import computePopupPlacement, { Alignment, Position, Placement } from './positioningHelper'
 
 const POSITIONS: Position[] = ['above', 'below', 'before', 'after']
 const ALIGNMENTS: Alignment[] = ['top', 'bottom', 'start', 'end', 'center']
 
 export interface IPopupProps {
   accessibility?: Accessibility
   align?: Alignment
-  as?: any
-  basic?: boolean
   children?: ReactChildren
   className?: string
   content?: ItemShorthand | ItemShorthand[]
+  defaultOpen?: boolean
+  open?: boolean
+  onOpenChange?: () => void
```

Comment:
should't we expose the `open` flag here as well; what's the advantage of this on having 2 props as in SUIR:
`onOpen` and `onClose`?

Reply:
+1 for adding handlers onOpen and onClose that would be invoked based on the value of the open. It is much cleaner API for the user.

Reply:
any thoughts here?